### PR TITLE
Add CHANGELOG

### LINKED ISSUE #17: Add a CHANGELOG file
This project is missing a CHANGELOG file

### PR DESCRIPTION
**Pull request by whymarrh**
Closes #17

This PR adds a `CHANGELOG` file.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+## [Unreleased]
+
+## [4.1.0] - 2020-10-28
+
+### Changed
+
+- Migrate to TypeScript ([#27](https://github.com/MetaMask/obs-store/pull/27))
+
+[Unreleased]:https://github.com/MetaMask/obs-store/compare/v4.1.0...HEAD
+[4.1.0]:https://github.com/MetaMask/obs-store/compare/v4.0.3...v4.1.0
```
Comment (Gudahtt) [inline CHANGELOG.md:16]: Sadly there are no tags in this repo, so these links don't work

Comment (whymarrh) [reply CHANGELOG.md:16]: Ah yes, I was going to create those

Comment (whymarrh) [reply CHANGELOG.md]: I created the tags that are linked

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**